driver-core/src/test/java/com/datastax/driver/core/IndexMetadataTest.java

```diff
@@ -0,0 +1,200 @@
+package com.datastax.driver.core;
+
+import static com.datastax.driver.core.ColumnMetadata.COLUMN_NAME;
+import static com.datastax.driver.core.ColumnMetadata.COMPONENT_INDEX;
+import static com.datastax.driver.core.ColumnMetadata.INDEX_NAME;
+import static com.datastax.driver.core.ColumnMetadata.INDEX_OPTIONS;
+import static com.datastax.driver.core.ColumnMetadata.INDEX_TYPE;
+import static com.datastax.driver.core.ColumnMetadata.KIND;
+import static com.datastax.driver.core.ColumnMetadata.VALIDATOR;
+import static com.datastax.driver.core.DataType.cint;
+import static com.datastax.driver.core.DataType.text;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.List;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.datastax.driver.core.ColumnMetadata.IndexMetadata;
+import com.datastax.driver.core.ColumnMetadata.Raw;
+import com.datastax.driver.core.Token.M3PToken;
+import com.datastax.driver.core.utils.CassandraVersion;
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Ints;
+
+@CassandraVersion(
+    major = 2.1)
+public class IndexMetadataTest extends CCMBridge.PerClassSingleNodeCluster {
+
+    @Override
+    protected Collection<String> getTableDefinitions() {
+        String createTable = "CREATE TABLE indexing ("
+            + "id int primary key,"
+            + "map_values map<text, int>,"
+            + "map_keys map<text, int>,"
+            + "map_entries map<text, int>,"
+            + "map_full frozen<map<text, int>>,"
+            + "set_full frozen<set<text>>,"
+            + "list_full frozen<list<text>>,"
+            + "custom text)";
+        return ImmutableList.of(createTable);
```

Comment:
It would be nice to have at least one simple test for Cassandra 1.2.x. This will be a bit cumbersome because you'll have to avoid creating tables with 2.1 features here, but you can test the version this way:

```java
        VersionNumber version = VersionNumber.parse(System.getProperty("cassandra.version"));
        if (version.compareTo(VersionNumber.parse("2.1.0")) >= 0) {
            ...
        }
```